lib/commandCenter/commandCenter.js

```diff
@@ -388,15 +387,52 @@ function compress(content, cb) {
 	});
 }
 
+/**
+ * Promisify the user command execute method, or
+ * ensure its return value is passed to state.respond
+ *
+ * Once callback-based user commands are removed from MAGE,
+ * you will be able to remove this method.
+ *
+ * @param {*} cmdInfo
+ * @param {*} paramList
+ */
+function generateCommandPromise(cmdInfo, state, paramList) {
+	const mod = cmdInfo.mod;
+	const execute = mod.execute.apply.bind(mod.execute, mod, paramList);
+
+	if (cmdInfo.isAsync) {
+		// Async user commands will need to take the response value, and feed it
+		// to `state.respond`
+		return execute()
+			.then((response) => state.respond(response, mod.serialize === false));
+	} else {
+		// Callback-based user commands are expected to call `state.respond`
+		// on their own; all we need to do is promisify
+		return new Promise((resolve) => mod.execute(resolve));
+	}
+}
 
 /**
- * Sets the execution timeout of all user commands to the given duration
+ * Create a promise timer
+ *
+ * Useful when using Promise.race.
  *
- * @param {number} timeout   Timeout in milliseconds
+ * @param {*} timeout
```

Comment:
Why remove all typing?

Reply:
I think this diff compares lines between a function that was removed and a new function but will add the type information.